extensions/contract-termination/src/test/java/de/sovity/edc/extension/contacttermination/query/ContractContractAgreementTerminationDetailsQueryTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ *  Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.extension.contacttermination.query;
+
+import de.sovity.edc.ext.db.jooq.enums.ContractTerminatedBy;
+import de.sovity.edc.extension.contacttermination.ContractAgreementTerminationDetails;
+import de.sovity.edc.extension.e2e.db.EdcRuntimeExtensionWithTestDatabase;
+import lombok.val;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiationStates;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+import static de.sovity.edc.extension.e2e.connector.config.ConnectorConfigFactory.forTestDatabase;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+class ContractContractAgreementTerminationDetailsQueryTest {
+
+    @RegisterExtension
+    static EdcRuntimeExtensionWithTestDatabase providerExtension = new EdcRuntimeExtensionWithTestDatabase(
+        ":launchers:connectors:sovity-dev",
+        "edc",
+        testDatabase -> {
+            val config = forTestDatabase("my-edc-participant-id", testDatabase);
+            return config.getProperties();
+        }
+    );
+
+    @Test
+    void fetchAgreementDetails_whenAgreementIsPresent_shouldReturnTheAgreementDetails(DSLContext dsl) {
+        // arrange
+
+        dsl.transaction(trx -> {
+                dsl.execute(
+                    new String(
+                        ContractContractAgreementTerminationDetailsQueryTest.class
+                            .getResource("/sql/AgreementTerminationDetailsQueryTest/init.sql").openStream().readAllBytes()
+                    ));
+
```

Comment:
i feel like this should move into a test utils class just for the class of readability

```java
TestSqlUtils.executeSql(dsl, "/sql/details-query-test.sql", getClass())
```

Also, this pattern is not very future-friendly, it will be as SQL scripts, that will not be migrated with future migrations, or as code that could be DRY-ed and fixed obviously

I'd argue that SQL inserts are shitty to keep in general. I mean you could simply create the records, since you have JooQ, so why use SQL